mandible/metadata_mapper/storage.py

```diff
@@ -1,57 +1,64 @@
 import re
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
-from typing import IO, ClassVar, Dict, Optional, Union
+from dataclasses import dataclass, field
+from typing import IO, Any, Dict, Type
 
 import s3fs
 
 from .context import Context
 
 
+class StorageError(Exception):
+    pass
+
+
+STORAGE_REGISTRY: Dict[str, Type["Storage"]] = {}
+
+
 @dataclass
 class Storage(ABC):
     # Registry boilerplate
-    _SUBCLASSES: ClassVar[Dict[str, "Storage"]] = {}
-
     def __init_subclass__(cls):
-        Storage._SUBCLASSES[cls.__name__] = cls
-
-    @classmethod
-    def get_subclass(cls, name: str) -> "Storage":
-        return cls._SUBCLASSES[name]
+        STORAGE_REGISTRY[cls.__name__] = cls
 
     # Begin class definition
-    name: Optional[str] = None
-    name_match: Optional[Union[str, re.Pattern]] = None
+    key_map: Dict[str, str] = field(default_factory=dict)
+    """A mapping for telling the Storage which keys to use for location
+    information."""
```

Comment:
Not sure if docstrings for attributes go above or below... Maybe a comment would be better?

Reply:
Oh this was actually an idea that I ended up not implementing (yet), so it can be removed.